python/documentation.md

```diff
@@ -0,0 +1,165 @@
+
+```python
+class PostList(PostResource):
+
+    def post(self):
+        ...
+        post_id = str(uuid.uuid1())
+        values = dict(
+            id = post_id,
+            created = datetime.now().isoformat(),
+            user = data.user,
+            text = data.text,
+            image_ref = data.image_ref,
+            like_count = 0,
+        )
+        k = list(values.keys())
+        v = list(values.values())
+        self.cursor.execute("""INSERT INTO {} ({}) VALUES ({})""".format(
+            self.__table__,
+            ', '.join(k),
+            ', '.join('?' * len(v))
+        ), v)
+        ...
+```
+
+#### Executing multiple statements (bulk operations)
+
+The same way as it is possible to use parameter substitution in a single execute statement it is also possible to execute multiple statements at once. This can be done using the `executemany()` method. The second argument requires a list of tuples where each tuple represents the parameter of a single statement.
+
+```python
+cursor.execute("""
+	INSERT INTO guestbook.posts (
+		id, text, user, image_ref
+	) VALUES (
+		?, ?, ?, ?
+	)""", [
+		(id1, text1, user1, image_ref1,),
+		(id2, text2, user2, image_ref2,),
+		...
+	])
+```
+
+### Fetching query results
+
+To fetch data from row returning statements (DQL statements) there are multiple methods: `fetchone()` (fetch next row of result set), `fetchmany()` (fetch next x rows of result set) and `fetchall()` (fetch all remaining rows of result set). These methods can be called after `execute()` to retrieve rows. For a detailed documentation see [PEP 0249 Cursor methods](https://www.python.org/dev/peps/pep-0249/#cursor-methods).
+
+```python
+cursor.execute("""
+	SELECT COUNT(*) FROM sys.nodes""")
+row = cursor.fetchone()
+```
+
+Our example application uses the data obtained from `fetchall()` and the `description` (which holds the column names) from the cursor and converts it into a list of dicts - which is then JSON serializable.
+
+```python
+class PostList(PostResource):
+
+    def convert(self, description, results):
+        cols = [c[0] for c in description]
+        return [dict(zip(cols, r)) for r in results]
+
+    def post(self):
+        post_id = str(uuid.uuid1())
+        ...
+        self.cursor.execute("""
+            SELECT p.*, c.name as country, c.geometry as area
+            FROM guestbook.posts AS p, guestbook.countries AS c
+            WHERE within(p.user['location'], c.geometry)
+              AND p.id = ?
+        """, (post_id,))
+        response = self.convert(self.cursor.description,
+                                self.cursor.fetchall())
+        ...
+```
+
+### Closing the connection
+
+Both the connection and the cursor can be closed on demand, rather then whenever `.__del__()` is called. Cursors and connections that have been closed cannot be used any more and will raise an error if any operation is attempted.
+
+```python
+cursor.close()
+connection.close()
+```
```

Comment:
i think we should also write a paragraph about blobs as soon as we have to blob stuff implemented